client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+        RecordedRequest redirectedRequest = server.takeRequest();
+        assertThat(redirectedRequest.getHeader(AUTHORIZATION)).isEqualTo("Bearer " + DUMMY_ACCESS_TOKEN);
+    }
+
+    @Test
+    public void testClientRedirectDisabled()
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect_host/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        try {
+            QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.of(true)), createClientSession(server));
+
+            try (Query query = queryRunner.startQuery("SELECT 1")) {
+                query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
+            }
+        }
+        catch (Exception e) {
+            assertThat(e).isInstanceOf(ClientException.class);
+            assertThat(e.getMessage()).isEqualTo("Client redirection was detected, but 'disableFollowRedirects' flag is set to true");
+        }
+    }
+
+    static TrinoUri createTrinoUri(MockWebServer server, Optional<Boolean> disableFollowRedirects)
+            throws SQLException
+    {
+        Properties properties = new Properties();
+        if (disableFollowRedirects.isPresent()) {
+            properties.setProperty(PropertyName.DISABLE_FOLLOW_REDIRECTS.toString(), Boolean.toString(disableFollowRedirects.get()));
+        }
+        properties.setProperty(PropertyName.ACCESS_TOKEN.toString(), DUMMY_ACCESS_TOKEN);
+        properties.setProperty(PropertyName.SSL.toString(), Boolean.toString(true));
+        return TrinoUri.create(server.url("/").uri(), properties);
+    }
+
+    static String createResults(MockWebServer server)
+    {
+        QueryResults queryResults = new QueryResults(
+                "20240327_104103_00010_pu6b8",
+                server.url("/query.html?20240327_104103_00010_pu6b8").uri(),
+                null,
+                null,
+                ImmutableList.of(new Column("_col0", BIGINT, new ClientTypeSignature(BIGINT))),
+                ImmutableList.of(ImmutableList.of(1)),
+                StatementStats.builder()
+                        .setState("FINISHED")
+                        .setProgressPercentage(OptionalDouble.empty())
+                        .setRunningPercentage(OptionalDouble.empty())
+                        .build(),
+                null,
+                ImmutableList.of(),
+                null,
+                null);
+        return QUERY_RESULTS_CODEC.toJson(queryResults);
+    }
+
+    static QueryRunner createQueryRunner(TrinoUri uri, ClientSession clientSession)
+    {
+        return new QueryRunner(
+                uri,
+                clientSession,
+                false,
+                HttpLoggingInterceptor.Level.NONE);
+    }
+
+    static PrintStream nullPrintStream()
+    {
+        return new PrintStream(nullOutputStream());
+    }
+}
```

Comment:
I would like another test for multiple redirection.

Reply:
`testClientRedirectsMultipleHop`